pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/FileExaminer.java

```diff
@@ -0,0 +1,110 @@
+package sfdc.sfdx.scanner.pmd;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarInputStream;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/***
+ * Examines files and helps scout for jar files and XMLs
+ * @author rmohan
+ * Feb 2020
+ */
+public class FileExaminer {
```

Comment:
Could you rename this to something like `XmlFileFinder`? `FileExaminer` implies that it examines the contents of the files, when it's really just identifying files that are XMLs.

Reply:
It also looks for Jar files. And this class also makes sure the files exist and checks for directories. There's one piece of code that would recursively walk through the contents of the directory and pick things of interest.

Reply:
But the only public-facing methods are about finding XMLs, right? Or am I missing something?

Reply:
You are right. I'm renaming this class.